src/node.rs

```diff
@@ -550,6 +550,12 @@ impl<'a> NodeProperty<'a> {
         core::str::from_utf8(self.value).map(|s| s.trim_end_matches('\0')).ok()
     }
 
+    /// Attempts to parse the property value as a list of [`&str`].
+    pub fn as_str_list(self) -> Option<impl Iterator<Item = &'a str> + 'a> {
+        let mut s = self.as_str()?.split('\0');
+        Some(core::iter::from_fn(move || s.next()))
```

Comment:
This could be simplified down to `Some(self.as_str()?.split('\0'))`, I believe

Reply:
> This could be simplified down to Some(self.as_str()?.split('\0')), I believe

No, because the `?` operator cannot be used in a function that doesn't return an `Option/Result` type.

Reply:
> This could be simplified down to Some(self.as_str()?.split('\0')), I believe

The `Split` iterator returned does not match the `impl Iterator<Item = &'a str> + 'a` return type. I tried this before, and using the `core::iter::from_fn` was the cleanest implementation that works, and stays consistent with the rest of the library.

Reply:
The reason I implemented it this way is because removing the `Option` wrapper from the return type requires separating the iterator creation.

I force-pushed a commit with the version of this function that returns a bare `impl Iterator` type.